src/ncp/thread_host.hpp

```diff
@@ -79,6 +102,14 @@ class ThreadHost
                                               bool                             aDryRun,
                                               bool                             aEnableAutoAttach);
 
+    /**
+     * Returns the Network properties.
+     *
+     * @returns the Network properties.
+     *
+     */
+    virtual const NetworkProperties &GetNetworkProperties(void) = 0;
```

Comment:
Could we make the `NetworkProperties` as the base class of the `ThreadHost`? So that we can directly call the function like `mHost->GetDeviceRole();` rather than `mHost->GetNetworkProperties().GetDeviceRole();`.

Reply:
Thanks! Have updated.